Chatto/sources/ChatController/ChatMessages/ChatMessageCollectionAdapter.swift

```diff
@@ -0,0 +1,702 @@
+//
+// Copyright (c) Badoo Trading Limited, 2010-present. All rights reserved.
+//
+
+import UIKit
+
+public protocol ChatMessageCollectionAdapterProtocol: UICollectionViewDataSource, UICollectionViewDelegate {
+    var chatItemCompanionCollection: ChatItemCompanionCollection { get }
+    var delegate: ChatMessageCollectionAdapterDelegate? { get set }
+
+    func startProcessingUpdates()
+    func stopProcessingUpdates()
+    func setup(in collectionView: UICollectionView)
+
+    // TODO: Remove from the adapter
+    func indexPath(of itemId: String) -> IndexPath?
+    func refreshContent(completionBlock: (() -> Void)?)
+    func scheduleSpotlight(for: String)
+}
+
+public protocol ChatMessageCollectionAdapterDelegate: AnyObject {
+    var isFirstLoad: Bool { get }
```

Comment:
What do you think about making the adapter to be owner of this state?

Reply:
That is a good idea!
Based on the previous logic, this state is being changed on the first `viewDidLayoutSubviews`. We could make the VC call a method called, p.e. `startAnimatingItemChanges`, in the adapter and change the `.firstLoad` nomenclature so is not so attached to the VC semantic.
If you are ok for that, I would make this small refactoring in a future PR.